interfaces/apparmor/spec.go

```diff
@@ -286,43 +323,21 @@ func (spec *Specification) AddLayout(si *snap.Info) {
 		sort.Strings(spec.snippets[tag])
 	}
 
-	emit := spec.AddUpdateNSf
-
 	// Append update-ns snippets that allow constructing the layout.
 	for _, path := range paths {
-		l := si.Layout[path]
-		emit("  # Layout %s\n", l.String())
-		path := si.ExpandSnapVariables(l.Path)
-		switch {
-		case l.Bind != "":
-			bind := si.ExpandSnapVariables(l.Bind)
-			// Allow bind mounting the layout element.
-			emit("  mount options=(rbind, rw) \"%s/\" -> \"%s/\",\n", bind, path)
-			emit("  mount options=(rprivate) -> \"%s/\",\n", path)
-			emit("  umount \"%s/\",\n", path)
-			// Allow constructing writable mimic in both bind-mount source and mount point.
-			GenWritableProfile(emit, path, 2) // At least / and /some-top-level-directory
-			GenWritableProfile(emit, bind, 4) // At least /, /snap/, /snap/$SNAP_NAME and /snap/$SNAP_NAME/$SNAP_REVISION
-		case l.BindFile != "":
-			bindFile := si.ExpandSnapVariables(l.BindFile)
-			// Allow bind mounting the layout element.
-			emit("  mount options=(bind, rw) \"%s\" -> \"%s\",\n", bindFile, path)
-			emit("  mount options=(rprivate) -> \"%s\",\n", path)
-			emit("  umount \"%s\",\n", path)
-			// Allow constructing writable mimic in both bind-mount source and mount point.
-			GenWritableFileProfile(emit, path, 2)     // At least / and /some-top-level-directory
-			GenWritableFileProfile(emit, bindFile, 4) // At least /, /snap/, /snap/$SNAP_NAME and /snap/$SNAP_NAME/$SNAP_REVISION
-		case l.Type == "tmpfs":
-			emit("  mount fstype=tmpfs tmpfs -> \"%s/\",\n", path)
-			emit("  mount options=(rprivate) -> \"%s/\",\n", path)
-			emit("  umount \"%s/\",\n", path)
-			// Allow constructing writable mimic to mount point.
-			GenWritableProfile(emit, path, 2) // At least / and /some-top-level-directory
-		case l.Symlink != "":
-			// Allow constructing writable mimic to symlink parent directory.
-			emit("  \"%s\" rw,\n", path)
-			GenWritableProfile(emit, path, 2) // At least / and /some-top-level-directory
-		}
+		layout := si.Layout[path]
+		spec.emitLayout(si, layout)
+	}
+}
+
+// AddExtraLayouts adds additional apparmor snippets based on the provided layouts.
+// The function is in part identical to AddLayout, except that it directly emits
+// the extra layouts.
```

Comment:
nitpick: I'd suggest a minor reformulation: "expect that it considers only the layouts passed as parameters instead of those declared in the snap.Info structure.".

Reply:
Thank you for your review! And I think your suggestion makes a lot of sense, I'll add it!